src/storage/redis_db.cc

```diff
@@ -366,6 +368,10 @@ rocksdb::Status Database::Scan(engine::Context &ctx, const std::string &cursor,
 
       if (metadata.Expired()) continue;
       std::tie(std::ignore, user_key) = ExtractNamespaceKey<std::string>(iter->key(), storage_->IsSlotIdEncoded());
+
+      if (!util::StringMatch(suffix_glob, user_key.substr(prefix.size()))) {
+        continue;
+      }
       keys->emplace_back(user_key);
       cnt++;
```

Comment:
I'm wondering that, if nothing got matched in this limit, e.g. the limit is 10, and in this 10 keys no key is matched, but the 12th key is matched, will a valid cursor be returned so that users can use it to continue the scan?

Reply:
it seems like `cnt` is only incremented when we actually add a key to the result, so the loop will continue until no more keys match the prefix, or `limit` keys are inserted into the result vector

Reply:
Hmmm I think we'd better refactor the logic, so that the SCAN can be quick, instead of a long-time blocking scan. E.g. we can scan a fixed maximum numbers of keys even if the limit/cnt is not reach.

Reply:
I'm not sure if we can just merge this first and then plan to refactor to a "quick" scan.

WDYT? cc @mapleFU @git-hulk

Reply:
Sorry for missing this comment. This sounds good to me after taking a rough review.

> Hmmm I think we'd better refactor the logic, so that the SCAN can be quick, instead of a long-time blocking scan. E.g. we can scan a fixed maximum numbers of keys even if the limit/cnt is not reach. It's fine to return even zero matched key to users.
Or maybe we can confirm the logic in Redis?

Yes, Redis will set a max iteration number(10*count) to avoid blocking too long.

Refer: https://github.com/redis/redis/blob/611c950293ae34dcef148ec62c9dd9626d7dc9e3/src/db.c#L1212